modules/serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/JobDispatcher.java

```diff
@@ -0,0 +1,646 @@
+   */
+  private List<String> undispatchableJobTypes = null;
+
+  /** The dispatcher priority list */
+  protected final Map<Long, String> dispatchPriorityList = new HashMap<>();
+
+  /** OSGi DI */
+  @Reference(target = "(osgi.unit.name=org.opencastproject.common)")
+  void setEntityManagerFactory(EntityManagerFactory emf) {
+    this.emf = emf;
+  }
+
+  /** OSGi DI */
+  @Reference()
+  void setServiceRegistry(ServiceRegistryJpaImpl sr) {
+    this.serviceRegistry = sr;
+  }
+
+  @Reference()
+  void setOrganizationDirectoryService(OrganizationDirectoryService organizationDirectoryService) {
+    this.organizationDirectoryService = organizationDirectoryService;
+  }
+
+  @Reference
+  void setUserDirectoryService(UserDirectoryService svc) {
+    this.userDirectoryService = svc;
+  }
+
+  @Reference
+  void setSecurityService(SecurityService sec) {
+    this.securityService = sec;
+  }
+
+  @Reference
+  void setTrustedHttpClient(TrustedHttpClient client) {
+    this.client = client;
+  }
+
+  @Activate
+  public void activate(ComponentContext cc) throws ConfigurationException  {
+    logger.info("Activate job dispatcher");
+    scheduledExecutor = (ScheduledThreadPoolExecutor) Executors.newScheduledThreadPool(1);
+    scheduledExecutor.setRemoveOnCancelPolicy(true);
+    logger.info("Activated");
+    updated(cc.getProperties());
+  }
+
+
+  @Modified
+  public void modified(ComponentContext cc) throws ConfigurationException {
+    logger.debug("Modified in job dispatcher");
+    updated(cc.getProperties());
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.osgi.service.cm.ManagedService#updated(java.util.Dictionary)
+   */
+  @SuppressWarnings("rawtypes")
+  public void updated(Dictionary properties) {
+
+    logger.info("Updating job dipatcher properties");
+
+    // NB: Read the javadoc for Boolean.valueOf.  This basically defaults to false.
+    boolean enableDispatch = Boolean.valueOf(StringUtils.trimToNull((String) properties.get(OPT_DISPATCHENABLED)));
+
+    long dispatchInterval = DEFAULT_DISPATCH_INTERVAL;
+    String dispatchIntervalString = StringUtils.trimToNull((String) properties.get(OPT_DISPATCHINTERVAL));
+    if (StringUtils.isNotBlank(dispatchIntervalString)) {
+      try {
+        dispatchInterval = Long.parseLong(dispatchIntervalString);
+      } catch (Exception e) {
+        logger.warn("Dispatch interval '{}' is malformed, setting to {}", dispatchIntervalString, MIN_DISPATCH_INTERVAL);
+        dispatchInterval = MIN_DISPATCH_INTERVAL;
+      }
+      if (dispatchInterval == 0) {
+        logger.info("Dispatching disabled");
+      } else if (dispatchInterval < MIN_DISPATCH_INTERVAL) {
+        logger.warn("Dispatch interval {} ms too low, adjusting to {}", dispatchInterval, MIN_DISPATCH_INTERVAL);
+        dispatchInterval = MIN_DISPATCH_INTERVAL;
+      } else {
+        logger.info("Dispatch interval set to {} ms", dispatchInterval);
+      }
+    }
+
+    long dispatchDelay = dispatchInterval;
```

Comment:
Wasn't this something like 30 seconds before? I remember UCT having problems with their Apache setup and wanting a certain waiting period before Opencast would start dispatching jobs. Does it make sense to have a configuration option for this? That way, if you don't specifically need it, you can have no delay?

Reply:
There wasn't (IIRC) a way to set that initial interval, but it's certainly reasonable to add such a thing.  @smarquard do you remember that issue?